core/tests/fixtures/fixture.ts

```diff
@@ -1,15 +1,22 @@
 import { type Page, type TestInfo } from '@playwright/test';
 
-import { TestApiClient } from '~/tests/fixtures/utils/api';
+import { testEnv } from '~/tests/environment';
+import { ApiClient, httpApiClient } from '~/tests/fixtures/utils/api';
 
 export abstract class Fixture {
-  protected readonly api: TestApiClient;
+  protected readonly api: ApiClient;
 
   constructor(
     readonly page: Page,
     readonly test: TestInfo,
   ) {
-    this.api = new TestApiClient(test);
+    this.api = httpApiClient;
+  }
+
+  protected skipIfReadonly(): void {
+    if (testEnv.TESTS_READ_ONLY) {
+      this.test.skip(true, 'Tests are running in read-only mode.');
```

Comment:
Can we see an example of this being skipped? I'm just curious how the test message will look because I though providing `tests are running in read-only mode.` overrides the test name 🤔

Reply:
It doesn't override the test name 👍  It is only used to give more information about why the test was skipped in the reporting.
Here's the run result:
![image](https://github.com/user-attachments/assets/bbeb1d80-74a2-49f0-a550-00dd8ff5f214)

And the reporting result:
![image](https://github.com/user-attachments/assets/8a704c87-40c8-41c4-b003-97e92b9da71b)